src/System.Management.Automation/engine/lang/parserutils.cs

```diff
@@ -750,23 +750,22 @@ private static object SplitWithPattern(ExecutionContext context, IScriptExtent e
                 separatorPattern = Regex.Escape(separatorPattern);
             }
 
-            if (limit < 0)
-            {
-                // Regex only allows 0 to signify "no limit", whereas
-                // we allow any integer <= 0.
-                limit = 0;
-            }
-
             RegexOptions regexOptions = parseRegexOptions(options);
+            // if the limit is negative then set Regex to read from right to left
```

Comment:
Nit: we should add a newline here for readability.
```suggestion

            // if the limit is negative then set Regex to read from right to left
```